### PR TITLE
feat(purgecss): additional options for purgecss

### PR DESCRIPTION
**Pull request by infoxicator**
This requires a new version of `purgecss-loader` once this PR is merged https://github.com/americanexpress/purgecss-loader/pull/9

- Added `purgecss` options to the common `webpack` loaders.
- Default options added as per the `purgecss` options documentation
- Added `whitelistPatternsChildren` default option as `[/:global$/]` so it doesn't remove global css wrapped in `:global {}` however this regular expression can be customised by module developers as per their needs.

Tested against `one-docs-root` and css is removed from the `<style>` elements in the header when classes are not used and classes wrapped in `:global {}` remain in the final sytlesheet

### CODE REVIEW

File: packages/one-app-bundler/__tests__/webpack/loaders/common.spec.js [hunk 1]
```diff
@@ -69,6 +69,24 @@ describe('Common webpack loaders', () => {
...
 
+    it('should include any additional configured options', () => {
+      const purgecss = {
+        paths: ['foo', 'bar'],
+        extractors: [{
+          extractor: 'purgeJs',
+          extensions: ['js'],
+        }],
+        fontFace: true,
+        keyframes: true,
```
Comment (nellyk) [inline packages/one-app-bundler/__tests__/webpack/loaders/common.spec.js:80]: Could you please add this to the README.md

Comment (infoxicator) [reply packages/one-app-bundler/__tests__/webpack/loaders/common.spec.js:80]: added in https://github.com/americanexpress/one-app-cli/pull/61/commits/aabe8f09c1e106c906c1d3c4897423f54ceadfc8 🌮

File: packages/one-app-bundler/README.md [hunk 2]
```diff
@@ -164,6 +164,37 @@ under `bundler.purgecss.paths`. The example below illustrates how we would add
...
+before enabling any of the following:
+
+<!-- #TODO: How to pass regex instead of string and how to pass a library to the extractor instead of string like "purgeJs" -->
+
+```json
+{
+  "one-amex": {
+      "bundler": {
+        "purgecss": {
+        "paths": ["node_modules/some-lib/src/**/*.{js,jsx}"],
```
Comment (JAdshead) [inline packages/one-app-bundler/README.md:177]: can you check the indentation

Comment (infoxicator) [reply packages/one-app-bundler/README.md]: solved here https://github.com/americanexpress/one-app-cli/pull/61/commits/3eea2fc005ac7c4257602f054509398258cb9610

File: packages/one-app-bundler/README.md [hunk 3]
```diff
@@ -164,6 +164,35 @@ under `bundler.purgecss.paths`. The example below illustrates how we would add
...
+        ]
+      }],
+      "fontFace": false,
+      "keyframes": false,
+      "variables": false,
+      "whitelist": ["random",
+        "yep",
+        "button"],
+      "whitelistPatterns": ["red"],
+      "whitelistPatternsChildren": ["blue"]
```
Comment (10xLaCroixDrinker) [inline packages/one-app-bundler/README.md:189]:
```suggestion
        "paths": ["node_modules/some-lib/src/**/*.{js,jsx}"],
        "extractors": [{
          "extractor": "purgeJs",
          "extensions": [
            "js"
          ]
        }],
        "fontFace": false,
        "keyframes": false,
        "variables": false,
        "whitelist": [
          "random",
          "yep",
          "button"
        ],
        "whitelistPatterns": ["red"],
        "whitelistPatternsChildren": ["blue"]
```

Comment (infoxicator) [reply packages/one-app-bundler/README.md]: I don't really know what went on with the indentation there 🤨

**Pull request merged**